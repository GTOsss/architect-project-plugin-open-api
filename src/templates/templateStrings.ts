import { createGeneratorStrByTemplate } from 'architect-project';

export const genCode = createGeneratorStrByTemplate({ itrStart: '~', itrEnd: '~' });

const fn = `export const ~name~ = (~params~) => ~body~` as const;

const inBrackets = `{ ~content~ }` as const;
const declarationType = `export type ~name~ = ~typeValue~` as const;

export const templates = { fn, inBrackets, declarationType };
