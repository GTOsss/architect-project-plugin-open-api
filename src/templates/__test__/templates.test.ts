import { genCode, templates } from '../templateStrings';

describe('Templates', () => {
  it('fn', () => {
    const result = genCode(templates.fn, { name: 'fnName', params: 'fnParams', body: '"some result"' });
    expect(result).toMatchSnapshot();
  });
});
