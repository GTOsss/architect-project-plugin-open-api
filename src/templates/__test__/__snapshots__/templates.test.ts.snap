// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates fn 1`] = `"export const fnName = (fnParams) => "some result""`;
